build_tools/benchmarks/run_benchmarks_on_android.py

```diff
@@ -183,17 +186,87 @@ def adb_start_cmd(
     return subprocess.Popen(cmd, stdout=subprocess.PIPE, text=True)
 
 
-def get_vmfb_full_path_for_benchmark_case(
-    benchmark_case_dir: pathlib.Path,
-) -> pathlib.Path:
-    flagfile = benchmark_case_dir / MODEL_FLAGFILE_NAME
-    for line in flagfile.read_text().splitlines():
-        flag_name, flag_value = line.strip().split("=")
-        if flag_name == "--module":
-            # Realpath canonicalization matters. The caller may rely on that to track
-            # which files it already pushed.
-            return (benchmark_case_dir / flag_value).resolve()
-    raise ValueError(f"{flagfile} does not contain a --module flag")
+def adb_path_exists(android_path: pathlib.PurePosixPath, verbose: bool = False):
+    """Run stat to check if the path exists."""
+    proc = adb_start_cmd(["stat", str(android_path)], verbose=verbose)
+    return proc.wait() == 0
+
+
+def adb_fetch_and_push_file(
+    source: Union[str, pathlib.Path],
+    dest: pathlib.PurePosixPath,
+    verbose: bool = False,
+):
+    """Fetch file from the path/URL and stream to the device.
+
+    In the case of fetching, this method avoids the temporary file on the host
+    and reduces the overhead when the file is large.
```

Comment:
Are the benchmark devices themselves connected to a network, or is this really streaming from the host's network through to adb? (I'd expect benchmark devices to be in airplane mode so security is better defined and benchmark execution has fewer external influences)

Reply:
The mobile phone currently don't have network access AFAIK, so here we stream the file through USB by adb.

Things might change later if it is actually okay to give phones the network access. We can always change the way to deliver files on the mobile phones later.